packages/xo-server-transport-icinga2/src/index.js

```diff
@@ -0,0 +1,126 @@
+import createLogger from '@xen-orchestra/log'
+import httpRequestPlus from 'http-request-plus'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:transport-icinga2')
+
+// =============================================================================
+
+export const configurationSchema = {
+  type: 'object',
+
+  properties: {
+    server: {
+      type: 'string',
+      description: 'The icinga2 server address',
+    },
+    port: {
+      type: 'integer',
+      description: 'The access port',
+    },
+    user: {
+      type: 'string',
+      description: 'The icinga2 server username',
+    },
+    password: {
+      type: 'string',
+      description: 'The icinga2 server password',
+    },
+    filter: {
+      type: 'string',
+      description: 'The filter to use',
+    },
+    rejectUnauthorized: {
+      type: 'boolean',
+      description: 'Reject unauthorized certificates',
+      default: true,
+    },
+  },
+  additionalProperties: false,
+  required: ['server', 'port', 'user', 'password', 'filter'],
+}
+
+// =============================================================================
+
+export const OK = 0
+export const WARNING = 1
+export const CRITICAL = 2
+export const UNKNOWN = 3
+
+// =============================================================================
+
+class XoServerIcinga2 {
+  constructor({ xo }) {
+    this._xo = xo
+  }
+
+  // ---------------------------------------------------------------------------
+
+  configure(configuration) {
+    const ENDPOINT = '/v1/actions/process-check-result'
+    this._url =
+      'https://' +
+      configuration.user +
+      ':' +
+      configuration.password +
+      '@' +
+      configuration.server +
+      ':' +
+      configuration.port +
+      ENDPOINT
+    this._filter = configuration.filter
+    this._rejectUnauthorized = configuration.rejectUnauthorized
+  }
+
+  load() {
+    this._unset = this._xo.defineProperty(
+      'sendIcinga2Status',
+      this._sendIcinga2Status,
+      this
+    )
+  }
+
+  unload() {
+    this._unset()
+  }
+
+  test() {
+    return this._sendIcinga2Status({
+      message:
+        'The server-icinga2 plugin for Xen Orchestra server seems to be working fine, nicely done :)',
+      status: OK,
+    })
+  }
+
+  // ---------------------------------------------------------------------------
+
+  async _sendIcinga2Status({ message, status }) {
+    try {
+      await httpRequestPlus
+        .post(this._url, {
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          options: {
+            rejectUnauthorized: this._rejectUnauthorized,
+          },
+          body: JSON.stringify({
+            type: 'Service',
+            filter: this._filter,
+            plugin_output: message,
+            exit_status: status,
+          }),
+        })
+        .readAll('utf8')
```

Comment:
You can remove the encoding param as you're not even using the output.

Reply:
so just `.readAll()`?

Reply:
Yes, it will return a raw buffer, but won't need to convert to an UTF-8 string.